src/pcp.rs

```diff
@@ -412,24 +412,21 @@ struct ProveShimGadget<F: FieldElement> {
 }
 
 impl<F: FieldElement> ProveShimGadget<F> {
-    #[allow(clippy::new_ret_no_self)]
-    fn new(
-        inner: Box<dyn Gadget<F>>,
-        g_calls: usize,
-    ) -> Result<Box<dyn Gadget<F>>, getrandom::Error> {
+    fn new(inner: Box<dyn Gadget<F>>, g_calls: usize) -> Result<Self, PcpError> {
```

Comment:
Change `g_calls` to `gadget_calls` for consistency with related changes elsewhere.